react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: ''
+    };
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({ username: event.target.value });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log(this.state);
+  }
+
+  render() {
+    return (
+      <form>
+        <label htmlFor="username">Username:</label>
+        <input type="text" id='username' name='username' value={this.state.username} onChange={this.handleUsernameChange}/>
+        <label htmlFor="password">Password:</label>
+        <input type="password" id='password' name='password' value={this.state.password} onChange={this.handlePasswordChange}/>
+        <button onClick={this.handleSubmit}>Sign up</button>
```

Comment:
Don't use `onClick` for submitting a form. Use the `onSubmit` event on the form element instead.
```JSX
<form onSubmit={this.handleSubmit}>
```

Reply:
![image](https://user-images.githubusercontent.com/43438792/189973645-4737517a-8542-49ac-9ddc-8359c89e60f9.png)